catalog/resource_grants.go

```diff
@@ -290,6 +290,18 @@ func parseId(d *schema.ResourceData) (string, string, error) {
 	return split[0], split[1], nil
 }
 
+// Check whether it's a first pass of forced new. From CustomizeDiff in schema.Resource:
+// "Existing resource, forced new: One run with state (before ForceNew), then one run without state (as if new resource)"
+func firstPassOfForcedNew(d *schema.ResourceDiff) bool {
+	fieldSetAsId := strings.Split(d.Id(), "/")[0]
```

Comment:
Let's reuse existing code rather than copying this here. Also, instead of `fieldSetAsId`, how about something like `securableType`?
```suggestion
	fieldSetAsId, _, _, _ := parseSecurableId(d.Id())
```